payload-examples/api.github.com/index.json

```diff
@@ -56415,6 +56415,34 @@
       }
     ]
   },
+  {
+    "name": "security_and_analysis",
+    "description": "Activity related to enabling or disabling code security and analysis features for a repository or organization.",
+    "properties": {
+      "changes": {
+        "type": "object",
+        "description": "The changes that were made to the code security and analysis features."
+      },
+      "repository": {
+        "type": "object",
+        "description": "The [`repository`](https://docs.github.com/en/rest/reference/repos#get-a-repository) where the event occurred."
+      },
+      "organization": {
+        "type": "object",
+        "description": "Webhook payloads contain the [`organization`](https://docs.github.com/en/rest/reference/orgs#get-an-organization) object when the webhook is configured for an organization or the event occurs from activity in a repository owned by an organization."
+      },
+      "installation": {
+        "type": "object",
+        "description": "The GitHub App installation. Webhook payloads contain the `installation` property when the event is configured for and sent to a GitHub App."
+      },
+      "sender": {
+        "type": "object",
+        "description": "The user that triggered the event."
+      }
+    },
+    "actions": [],
+    "examples": []
+  },
```

Comment:
@wolfy1339 looks like `security_and_analysis` is new? What's the process of adding types for it? I tried following instructions in `CONTRIBUTING.md`, but it's not clear about that. Is one of the scripts not working, or are the instructions incomplete? I'd appreciate if you could have a look

Reply:
I assume we need a payload example for `security_and_analysis`?

Reply:
> once we add the payload example, is there a script to help with creating the schema file or is it all done by hand?

Currently it's done by hand, iirc.

> What's the process of adding types for it?

The types are generated from the schemas, so once you've created a new schema you then should follow the [updating types](https://github.com/octokit/webhooks/blob/master/CONTRIBUTING.md#updating-types) section of the contributing doc.

I think what we're missing is a middle section titled "Adding new schemas" in the contributing doc.

Reply:
> Is there a reason the update scripts don't do that based on the example payload in the docs?

 It's just been set up this way since the beginnings of this project, and nobody has bothered to change it

> And once we add the payload example, is there a script to help with creating the schema file or is it all done by hand?

Making a schema using scripts could potentially be used as a base, but would always require going back to add more refined values, like `enum`s and `description`s, and then creating common schemas if needed

Reply:
Thank you both 👍🏼 I made a follow up issue to better document the current process at #643. I asked the @octokit team within GitHub if there are JSON Schemas for webhooks yet, and what the status is of adding webhooks/events schemas to the official OpenAPI at https://github.com/github/rest-api-description/, now that they publish version 3.1.

Reply:
good news: there are schemas for webhook payloads now within GitHub, but they are not published yet.

Reply:
That's really great to hear!